cmd/state-installer/cmd.go

```diff
@@ -482,3 +476,41 @@ func determineLegacyUpdate(stateToolInstalled bool, packagedStateExe, payloadPat
 func noArgs() bool {
 	return len(os.Args[1:]) == 0
 }
+
+var branchRegex = regexp.MustCompile("Branch (\\S+)")
+var versionRegex = regexp.MustCompile("Version (\\S+)")
+
+func shouldUpdateInstalledStateTool() bool {
+	logging.Debug("Checking if installed state tool is an older version.")
+
+	stateExe := constants.StateCmd + exeutils.Extension
+	cmd := exec.Command(stateExe, "--version")
```

Comment:
Please use one of the exeutils methods.

Reply:
I did, but they were not returning stdout :(